src/compositor/seat/pointer.c

```diff
@@ -63,13 +63,43 @@ view_visible(struct wlc_view *view, uint32_t mask)
    return (view->mask & mask);
 }
 
-static struct wlc_view*
-view_under_pointer(struct wlc_pointer *pointer, struct wlc_output *output)
+static void
+find_surface_at_position_recursive(struct wlc_pointer *pointer, struct wlc_surface *parent, struct wlc_focused_surface *out) {
+
+   struct wlc_surface *surface = parent;
+   wlc_resource *sub;
+
+   chck_iter_pool_for_each(&parent->subsurface_list, sub) {
+      struct wlc_surface *subsurface = convert_from_wlc_resource(*sub, "surface");
+      if(!subsurface) {
+         wlc_log(WLC_LOG_ERROR, "Invalid subsurface id %" PRIuWLC " parent %" PRIuWLC,
+                 convert_to_wlc_resource(subsurface), convert_to_wlc_resource(parent));
```

Comment:
This log is also unneccessary here. I can add such log into resource.c and make it toggable with WLC_DEBUG (return warning each time NULL is returned from handle lookup).